pkg/k8s/version/version.go

```diff
@@ -119,14 +125,29 @@ func updateServerGroupsAndResources(apiGroups []*metav1.APIGroup, apiResourceLis
 	defer cached.mutex.Unlock()
 
 	cached.capabilities.EndpointSlice = false
+	cached.capabilities.LeasesResourceLock = false
 	for _, rscList := range apiResourceLists {
 		if rscList.GroupVersion == discoveryAPIGroup {
 			for _, rsc := range rscList.APIResources {
 				if rsc.Kind == endpointSliceKind {
 					cached.capabilities.EndpointSlice = true
+					break
+				}
+			}
+		}
+
+		if rscList.GroupVersion == coordinationV1APIGroup {
+			for _, rsc := range rscList.APIResources {
+				if rsc.Kind == leaseKind {
+					cached.capabilities.LeasesResourceLock = true
+					break
 				}
 			}
 		}
+
+		if cached.capabilities.LeasesResourceLock && cached.capabilities.EndpointSlice {
+			break
+		}
```

Comment:
please add a comment in this line. To avoid mistakes I think we  should not even add this. In case in the future we add a new capability and forget to update this line.